rules/android_binary/impl.bzl

```diff
@@ -217,7 +217,18 @@ def _process_jvm(ctx, db_ctx, packaged_resources_ctx, proto_ctx, stamp_ctx, **_u
         javac_opts = ctx.attr.javacopts + db_ctx.javac_opts,
         r_java = packaged_resources_ctx.r_java,
         enable_deps_without_srcs = True,
-        deps = utils.collect_providers(JavaInfo, utils.dedupe_split_attr(ctx.split_attr.deps) + stamp_ctx.deps),
+        deps = (
+            utils.collect_providers(JavaInfo, utils.dedupe_split_attr(ctx.split_attr.deps) + stamp_ctx.deps) +
+            [
+                JavaInfo(
```

Comment:
This is a clever idea, but unfortunately it won't work in google3. The toolchain we use internally contains a bunch of other stuff on the bootclasspath besides the android.jar, so we'd need to add all of those as deps as well in order to maintain feature parity.

Furthermore, the desugaring step uses bootclasspath info from toolchains_android_only, so we do still need to keep some notion of a java toolchain with bootclasspath attr set.

This PR also adds android.jar to the DefaultInfo of every Android target. I'm not sure if that would necessarily cause a performance regression internally, but it certainly is a little unexpected.

We've made some changes recently to how the default_java_toolchain is constructed. I understand that for you and other OSS rules_android users that the java toolchain config there is problematic; maybe we could find another way to bridge the gap?